CONTRIBUTING.adoc

```diff
@@ -108,55 +108,61 @@ NOTE: you must run your editor from the nix shell, and not from, say, an
 application launcher like spotlight or dmenu for your editor to find it in
 its PATH.
 
-=== How to add a new Haskell package
-
-You need to do a few things when adding a new package, in the following order:
-
-. Add the cabal file for the new package.
-. Add the package to link:cabal.project[`cabal.project`].
-. Add the package to link:stack.yaml[`stack.yaml`].
-. Check that you can run `nix build -f default.nix plutus-apps.haskell.projectPackages.<package name>` successfully.
-
-[[update-haskell-pins]]
-=== How to update our pinned Haskell dependencies
+[[update-index-state]]
+=== Updating dependencies
 
-We have pinned versions of some Haskell packages specified via the usual `source-repository-package` (Cabal) and `extra-dep` (Stack) mechanisms.
+==== ... from Hackage
 
-These can be managed normally, but ensure that:
+Updating package dependencies from Hackage should work like normal in a
+Haskell project. The most important thing to note is that we pin the
+``index-state`` of the Hackage package index in ``cabal.project``. This
+means that cabal will always see Hackage “as if” it was that time, ensuring
+reproducibility. But it also means that if you need a package version that
+was released *after* that time, you need to bump the ``index-state`` (and
+to run ``cabal update`` locally). Please also note that index-state is a
+property of your working environment (the cabal project) not of the
+packages. This means that downstream consumers have no idea of your
+`index-state` setting and you need to make sure your packages work
```

Comment:
nit: single ` as opposed to double `` here and on line 126

Reply:
:pray: